pkgs/top-level/rust-packages.nix

```diff
@@ -1,49 +1,35 @@
-# This file defines the source of Rust / cargo's crates registry
-#
-# buildRustPackage will automatically download dependencies from the registry
-# version that we define here. If you're having problems downloading / finding
-# a Rust library, try updating this to a newer commit.
-
-{ runCommand, fetchFromGitHub, git }:
-
-let
-  version = "2016-06-26";
-  rev = "eb6b06424a5cdc5dd0f41f06cda69714b3a0577b";
-
-  src = fetchFromGitHub {
-      inherit rev;
-
-      owner = "rust-lang";
-      repo = "crates.io-index";
-      sha256 = "0iyqyyd5070whfgsg74bqk0f6rk7d2r4spb2ih9jy594ds2qinv4";
+{ lib, pkgs, callPackage, newScope }:
+
+rec {
+  makeRustPlatform = rust: lib.fix (self:
+    let
+      callPackage = newScope self;
+    in {
+      inherit rust;
+
+      rustRegistry = callPackage ../development/rust-modules { };
+
+      buildRustPackage = callPackage ../build-support/rust {
+        inherit rust;
+      };
+    });
+
+  tools = {
+    rust-1_9_0-binary = callPackage ../development/compilers/rust/1.9.0-binary.nix {};
+    rust-1_10_0 = callPackage ../development/compilers/rust/1.10.0.nix {
+      rustPlatform = platforms.rustPlatform-1_9_0-binary;
```

Comment:
Especially I don't think we should keep old versions around (except for bootstrapping, like `1.9.0-binary` here), so explicit versioning in the attribute-paths is quite pointless. Plus every attribute path with a version in it is a place that needs to be updated when a new version comes along, making updates more complicated.